scripts/check

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+
+file=
+ci=
+
+usage_message="Usage: check [OPTIONS]
+OPTIONS:
+--file=<filename>          Specify file to check. If none specified, all
+                           files in src/ and include/ will be checked.
+--ci                       Die on finding warnings/errors
+"
+
+run_checkpatch() {
+	if [ "$ci" = "t" ]; then
+		scripts/checkpatch.pl --terse --no-tree --file "$1" || exit 1
+	else
+		scripts/checkpatch.pl --terse --no-tree --strict --file "$1"
+	fi
+}
+
+run_checks () {
+	if [ ! -z "$file" ]; then
+		run_checkpatch "${file}"
+		return
+	fi
+
+	for f in `find src/ -name "*.c" -type f`; do
```

Comment:
I think there is some kind of limit on the amount of arguments like this.
So piping the result to a `while read -r file` loop might be more robust, possibly combined with having `find` use `\0` as separator and the same for the `while` loop.

Or alternatively we could use the execute option of find. However, I am not sure if those then run in parallel and if so, how many instances of those would be running concurrently.